avalon/tools/loader/model.py

```diff
@@ -141,17 +150,72 @@ def set_version(self, index, version):
             "step": version_data.get("step", None)
         })
 
-    def refresh(self):
+    def fetch_subset_and_version(self):
+        """Query all subsets and latest versions from aggregation
+
+        (NOTE) The returned version documents are NOT the real version
+            document, it's generated from the MongoDB's aggregation so
+            some of the first level field may not be presented.
+
+        """
+        def _fetch():
```

Comment:
`_fetch` should be class method, there is not much reason to have it inside.

Reply:
I guess it's there because it acts more like a local lambda written as a function. 

Also, if it were put on the class it shouldn't be a classmethod right? Because it uses variables/functions of the instance of the class inside of it (not how it refers to self). So if put on the class it would be a regular method.

Reply:
> Also, if it were put on the class it shouldn't be a classmethod right?

I meant as object method `def _fetch(self):`, not as class method, my fault.

> it acts more like a local lambda written as a function

I see advantage of lamda but there's no advantage of that here, all `_fetch` need is `self`. Method `fetch_subset_and_version` has 4 lines in fact. All what these lines do, is preparation and launching another method in thread, but because the definition of launched method is inside it's harder to read (for me) what's really happening inside. 

This is "formatting" comment of my preferences. I have no ambition to change mindset because I know it's hard to change mine :)

Reply:
Looks right to me. Nesting a function tells me that it is only useful to the parent function, which seems to be the case here. If you put it in the class or instance, I would expect it to be useful standalone and by others, and require its own tests. The smaller the interface the better. Also, if Python did support multiline lambdas, this would be a good place for it.